src/lib/y2network/dialogs/additional_address.rb

```diff
@@ -47,18 +47,42 @@ def contents
       end
 
       def run
-        ret = super
-        return ret if ret != :ok || @settings.subnet_prefix.start_with?("/")
+        ret = nil
 
-        netmask = @settings.subnet_prefix
-        prefix = IPAddr.new("#{netmask}/#{netmask}").prefix
-        @settings.subnet_prefix = "/#{prefix}"
+        loop do
+          ret = super
+          break if ret != :ok
+
+          @settings.subnet_prefix = subnet_prefix_for?(@settings.subnet_prefix)
+
+          begin
+            IPAddr.new("#{@settings.ip_address}#{@settings.subnet_prefix}")
+            break
+          rescue IPAddr::InvalidAddressError
+            Yast::Report.Error(
+              format(_("Invalid Address %s%s"), @settings.ip_address, @settings.subnet_prefix)
+            )
+          end
+        end
 
         ret
       end
 
     private
 
+      def subnet_prefix_for?(value)
```

Comment:
the question mark here looks really strange as it is not question, but normalization

Reply:
yep, you are right, I modified the code and it should be removed from current one.